core/vm/cvm.go

```diff
@@ -0,0 +1,42 @@
+package vm
+
+import (
+	"github.com/ledgerwatch/erigon/common"
+)
+
+func NewCVM(state IntraBlockState) *CVM {
+	cvm := &CVM{
+		intraBlockState: state,
+	}
+
+	return cvm
+}
+
+type CVM struct {
+	config          Config
+	intraBlockState IntraBlockState
+}
+
+func (cvm *CVM) Create(code []byte) ([]byte, common.Address, error) {
+	ret, err := cvm.run(code)
+
+	if err == nil {
+		address := common.Address{}
+		cvm.intraBlockState.SetCode(address, ret)
```

Comment:
probably it will be `SetCairoCode` or something like that (or a parameter) that uses a separate bucket.
it is additional work, but that separation is important to nail from the first time I think